ls1intum/Artemis
# `Programming exercises`: Fix missing feedback if test cases were not executed

### Checklist
#### General
#### Server
#### Client

### Motivation and Context
In #7322 we noticed that the 'Test not executed' feedback is not displayed in the feedback modal.

### Description
The positive attribute is not set for this type of feedback. If positive is undefined, we now use the credits to check if the test was sucessful or not.

### Steps for Testing

Prerequisites:
- 1 Instructor, 1 Student
- 1 Programming Exercise with sequential test runs

1. Ensure that the solution build ran at least once before the template build (see the discussion in #7322)
2. Check the template feedback and see that 'Test was not executed' is displayed for the behavioral tests.
3. Participate in the exercise as a student and check that this type of feedback is visible for students.

#### Exam Mode Testing
same as above but for exam exercises

### Review Progress
#### Performance Review
#### Code Review
#### Manual Tests
#### Exam Mode Test

### Test Coverage


### Screenshots
![grafik](https://github.com/ls1intum/Artemis/assets/26540346/d332c5bd-ab06-468e-b07d-2fcadb52014b)
